service/src/common/indexer_management_client/resolver.rs

```diff
@@ -0,0 +1,343 @@
+            },
+            |m| Some(merge_global(m, &global)),
+        )
+    } else {
+        model
+    };
+
+    Ok(model)
+}
+
+/// Query global cost model
+pub async fn global_cost_model(pool: &PgPool) -> Result<Option<CostModel>, IndexerError> {
+    let model = sqlx::query_as!(
+        CostModel,
+        r#"
+        SELECT deployment, model, variables
+        FROM "CostModels"
+        WHERE deployment = $1
+    "#,
+        "global"
+    )
+    .fetch_optional(pool)
+    .await
+    .map_err(|e| IndexerError::new(IndexerErrorCode::IE076, Some(IndexerErrorCause::new(e))))?;
+    Ok(model)
+}
+
+fn merge_global(model: CostModel, global_model: &CostModel) -> CostModel {
+    CostModel {
+        deployment: model.deployment,
+        model: model.model.clone().or(global_model.model.clone()),
+        variables: model.variables.clone().or(global_model.variables.clone()),
+    }
+}
+
+#[cfg(test)]
+mod test {
+
+    use sqlx::PgPool;
+
+    use super::*;
+
+    async fn setup_cost_models_table(pool: &PgPool) {
+        sqlx::query!(
+            r#"
+            CREATE TABLE "CostModels"(
+                id INT,
+                deployment VARCHAR NOT NULL,
+                model TEXT,
+                variables JSONB,
+                PRIMARY KEY( deployment )
+            );
+            "#,
+        )
+        .execute(pool)
+        .await
+        .expect("Create test instance in db");
+    }
+
+    async fn add_cost_models(pool: &PgPool, models: Vec<CostModel>) {
+        for model in models {
+            sqlx::query!(
+                r#"
+                    INSERT INTO "CostModels" (deployment, model)
+                    VALUES ($1, $2);
+                    "#,
+                model.deployment,
+                model.model,
+            )
+            .execute(pool)
+            .await
+            .expect("Create test instance in db");
+        }
+    }
+
+    fn global_cost_model() -> CostModel {
+        CostModel {
+            deployment: "global".to_string(),
+            model: Some("default => 0.00001;".to_string()),
+            variables: None,
+        }
+    }
+
+    fn simple_cost_models() -> Vec<CostModel> {
+        vec![
+            CostModel {
+                deployment: "0xbd499f7673ca32ef4a642207a8bebdd0fb03888cf2678b298438e3a1ae5206ea"
+                    .to_string(),
+                model: Some("default => 0.00025;".to_string()),
+                variables: None,
+            },
+            CostModel {
+                deployment: "0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+                    .to_string(),
+                model: None,
+                variables: None,
+            },
+        ]
+    }
+
+    #[sqlx::test]
+    #[ignore]
+    async fn success_cost_models(pool: PgPool) {
+        _ = setup_cost_models_table(&pool).await;
+        let expected_models = simple_cost_models();
+        _ = add_cost_models(&pool, expected_models.clone()).await;
```

Comment:
These functions do not return a value. Applies to all the tests below too.
```suggestion
        setup_cost_models_table(&pool).await;
        let expected_models = simple_cost_models();
        add_cost_models(&pool, expected_models.clone()).await;
```